gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14353,7 +14353,7 @@ private long getEntryOverhead(RegionEntry entry) {
     long entryOverhead = ReflectionSingleObjectSizer.INSTANCE.sizeof(entry);
     Object key = entry.getRawKey();
     if (key != null) {
-      entryOverhead += CachedDeserializableFactory.calcMemSize(entry);
+      entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(key);
     } else {
       // first key.
       Object firstKey = this.getRegionMap().keySet().iterator().next();
```

Comment:
Perhaps there was some confusion. My comment was to use CachedDeserializableFactory.calcMemSize(key) and CachedDeserializableFactory.calcMemSize(firstKey)  instead of ReflectionSingleObjectSizer since key type can implement Sizeable which latter won't be able to handle correctly. For entry overhead you need something like RelectionSingleObjectSizer.